### PR TITLE
Fetch comments on issue

### PR DESCRIPTION
**Pull request by Palleas**

### CODE REVIEW

File: Tentacle/Client.swift [hunk 1]
```diff
@@ -178,7 +178,10 @@ public final class Client {
 
         // https://developer.github.com/v3/issues/#list-issues-for-a-repository
         case IssuesInRepository(owner: String, repository: String)
-        
+
+        // https://developer.github.com/v3/issues/comments/#list-comments-on-an-issue
+        case CommentsOnIssue(number: String, owner: String, repository: String)
```
Comment (Palleas) [inline Tentacle/Client.swift]: Started using an instance of `Issue` and `Repository` here instead but it didn't feel right with the rest of the 
codebase... 🤔

Comment (mdiep) [inline Tentacle/Client.swift]: This looks good to me. 👍🏻

Comment (Palleas) [inline Tentacle/Client.swift]: Cool! Let's do this then! 😆

File: Tests/IssuesTests.swift [hunk 2]
```diff
@@ -16,7 +16,7 @@ class IssuesTests: XCTestCase {
         let palleasOpensource = User(
             ID: "15802020",
             login: "Palleas-opensource",
-            URL: NSURL(string: "https://api.github.com/users/Palleas-opensource")!,
+            URL: NSURL(string: "https://github.com/Palleas-opensource")!,
```
Comment (Palleas) [inline Tests/IssuesTests.swift:19]: Not sure why but it started to fail the tests when I started working on this 🤔

**Comment by Palleas** [comment]
I'm waiting for Travis to confirm but I think this is ready for a review. 🐮

File: Tentacle/Comment.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import Argo
+
+public struct Comment: Hashable, CustomStringConvertible {
+
+    public let ID: String
+    public let URL: NSURL
+    public let createdAt: NSDate
+    public let updatedAt: NSDate
+    public let body: String
+    public let user: User
```
Comment (mdiep) [inline Tentacle/Comment.swift]: These could use documentation—along with the struct itself. ☺️

**Comment by mdiep** [comment]
This looks great. It just needs a little API documentation. :sparkles:

File: Tentacle/Client.swift [hunk 4]
```diff
@@ -303,6 +310,11 @@ public final class Client {
         return fetchMany(.IssuesInRepository(owner: repository.owner, repository: repository.name), page: page, pageSize: perPage)
     }
 
+    public func commentsOnIssue(issue: Issue, repository: Repository, page: UInt = 1, perPage: UInt = 30) -> SignalProducer<(Response, [Comment]), Error> {
```
Comment (mdiep) [inline Tentacle/Client.swift:314]: This needs documentation too.

File: Tentacle/Comment.swift (hunk 3 above)
Comment (Palleas-opensource) [inline Tentacle/Comment.swift]: Considering I spent some times trying to generate documentation, the irony of me forgetting to document my code is... 🙊

**Comment by Palleas** [comment]
Documented! 🎉

File: Tentacle/Comment.swift [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    /// The URL to view this comment in a browser
+    public let URL: NSURL
+    /// The date this comment was created at
+    public let createdAt: NSDate
+    /// The date this comment was last updated at
+    public let updatedAt: NSDate
+    /// The body of the comment
+    public let body: String
+    /// The user who posted this comment
+    public let user: User
```
Comment (mdiep) [inline Tentacle/Comment.swift]: Ooo… can we actually change this to `author`? I think that's more descriptive.

Sorry I didn't catch this before. 🙈

Comment (Palleas) [inline Tentacle/Comment.swift]: Haha, I was wondering the same when I wrote this comment! Fixing right now!

Comment (Palleas) [inline Tentacle/Comment.swift]: Done! 🎉

**Comment by mdiep** [comment]
🎆

**Pull request merged**